eng/versioning/set_versions.py

```diff
@@ -175,18 +175,18 @@ def update_versions_file_for_nightly_devops(build_type, build_qualifier, artifac
                 # been modified.
                 elif (module.name.startswith('unreleased_') or module.name.startswith('beta_'))  and not module.dependency.startswith('['):
                     # Assuming a build qualifier of the form: "alpha.20200204.1"
-                    # Converts "alpha.20200204.1" -> "alpha.20200204."
-                    unreleased_build_qualifier = build_qualifier[:build_qualifier.rfind('.') + 1]
+                    # Converts "alpha.20200204.1" -> "alpha"
+                    unreleased_build_identifier = build_qualifier[:build_qualifier.find('.')]
+                    # Converts "alpha.20200204.1" -> "20200204.1"
+                    unreleased_build_date = build_qualifier[build_qualifier.find('.')+1:][:8]
 
                     if '-' in module.dependency:
-                        # if the module is 1.2.3-beta.x, strip off everything after the '-' and add the qualifier
-                        module.dependency = module.dependency[:module.dependency.rfind('-') + 1] + unreleased_build_qualifier
-                    else:
-                        module.dependency += '-' + unreleased_build_qualifier
+                        # if the module is 1.2.3-beta.x, strip off everything after the '-'
+                        module.dependency = module.dependency[:module.dependency.rfind('-')]
 
                     # The final unreleased dependency version needs to be of the form
-                    # [1.0.0-alpha.YYYYMMDD.,] <-- note the ., this is the version range for Maven
-                    module.dependency = '[{},]'.format(module.dependency)
+                    # [1.0.0-alpha.YYYYMMDD,1.0.0-alpha.99999999] this will keep it in alpha version of the same major/minor/patch range to avoid potential future breaks
```

Comment:
@alzimmermsft this change was needed because of the change in the versioning that has happen with azure-core-amqp (see comments in https://github.com/Azure/azure-sdk-for-java/pull/21232/files#r640266963). We essentially shipped beta versions of 2.1.0 and 2.2.0 which didn't have all the API's of the current 2.1.0-alpha.x because we recently added new APIs in that version which servicebus depended on with @YijunXieMS's PR https://github.com/Azure/azure-sdk-for-java/pull/21688. This caused a very interesting set of changes that causes servicebus to fail when using the latest dev versioning which ended up picking up the 2.2.0-beta.1 which didn't have the new API. For now I've decided to pin these unreleased versions to be in the same major/minor/patch -alpha range so that we can try an limit the scope of what sort of a breaking change. Ideally we wouldn't have any breaks like this so an open-ended range would work but that isn't the case here. 

We need to start thinking about our versioning a little more to ensure we avoid these type of breaks in the future.